### PR TITLE
Add Http3 kestrel doc

### PR DESCRIPTION
**Pull request by wtgodbe**
***@Rick-Anderson***  edit to add [internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?view=aspnetcore-6.0&branch=pr-en-us-22966)
Adds doc for requirements/getting started w/ Http/3 for preview7.

What sort of stuff should I include for getting started? Running the sample? Something like https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-5.0#get-started that specifies something like:

```
                        options.ListenAnyIP(5000, listenOptions =>
                        {
                            listenOptions.Protocols = HttpProtocols.Http3;
                        });
````

### CODE REVIEW

**Comment by wtgodbe** [comment]
Added a sample that probably needs cleaning up...

**Comment by JamesNK** [comment]
Need to add:

* Code sample that:
  * Sets `ListenOptions.Protocol` to `HttpProcols.Http1AndHttp2AndHttp3`.
  * Adds `UseQuic`
  * Set `EnableAltSvc` to `true`.
* How to check whether HTTP/3 is working. Should be able to browse to website using Chrome, open networking tab and see HTTP/3 in protocols column.

You should download preview 7 and verify the sample works in it.

**Comment by wtgodbe** [comment]
The sample runs w/ preview7 SDK, but I don't see anything HTTP/3 hitting `https://0.0.0.0:5557`

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (JamesNK) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:42]:
```suggestion
HTTP/3 is configured on app start-up. The following code:

* Configures the `WebHost` to `UseQuic`.
* Sets `EnableAltSvc` to `true` on Kestrel options.
* Configures port 5001 to use `HttpProtocols.Http1AndHttp2AndHttp3`.
```

Documentation pro-tip: people love lists.

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:31]:
```suggestion
On Linux, `libmsquic` is published via Microsoft official Linux package repository `packages.microsoft.com`. Add the repository following these instructions: https://docs.microsoft.com/en-us/windows-server/administration/linux-package-repository-for-microsoft-software. After that, you should be able to install libmsquic via the package manager of your distro, e.g. for Ubuntu:
```

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:38]:
```suggestion
HTTP/3 is not currently supported on macOS and may be available in a future release.
```

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:42]: We should add a note here that this code is specific to Preview7 and will change fore RC1.

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:31]:
```suggestion
On Linux, `libmsquic` is published via Microsoft official Linux package repository `packages.microsoft.com`. In order to consume it, it must be added manually. See [Linux Software Repository for Microsoft Products](/windows-server/administration/linux-package-repository-for-microsoft-software). After adding `libmsquic`, it can be installed via the package manager of your distro, for example, for Ubuntu:
```

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Startup.cs
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Startup.cs:25]:
```suggestion
```

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:15]:
```suggestion
[HTTP/3](https://quicwg.org/base-drafts/draft-ietf-quic-http.html) is available for ASP.NET Core apps when the following requirements are met:
```

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:21]:
```suggestion
Kestrel enables HTTP/3 only on environments that support it. That means it's possible to configure a port to support all HTTP protocols. For example, `HttpProtocols.Http1AndHttp2AndHttp3`, and Kestrel's HTTP/3 is available on environments where it's supported.
```

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:38]:
```suggestion
HTTP/3 support is planned on macOS in a future release.
```

**Review by Tratcher — COMMENTED** [review]
Should we add an alt-svc section?
- Http/3 is discovered as an upgrade from HTTP/1.1 or HTTP/2 via the alt-svc header. That means the first request will normally use HTTP/1.1 or HTTP/2 before switching to HTTP/3.

Also localhost testing:
- Browsers do not enable HTTP/3 on localhost/loopback connections, to test with a browser run the client and server on separate machines.
- HttpClient can be used for localhost/loopback testing. (link to HttpClient Http/3 docs?)

File: aspnetcore/fundamentals/servers/kestrel/http3.md
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:50]:
```suggestion
For more information on building the host, see the **Set up a host** and **Default builder settings** sections of <xref:fundamentals/host/generic-host#set-up-a-host>.
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs
Comment (JamesNK) [inline aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:20]: Change this to use minimal hosting.

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs
Comment (JamesNK) [inline aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:18]: Remove. Use default ASP.NET Core dev cert

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs
Comment (JamesNK) [inline aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:43]:
```suggestion
                            // Use Http3
                            listenOptions.Protocols = HttpProtocols.Http1AndHttp2AndHttp3;
                            listenOptions.UseHttps();
```

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs
Comment (JamesNK) [inline aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:35]:
```suggestion
                        options.EnableAltSvc = true;
                        options.Listen(IPAddress.Any, 5001, listenOptions =>
```

**Review by JamesNK — COMMENTED** [review]
When writing a code sample you should try trim out anything that is unnecessary.

**Comment by JamesNK** [comment]
Need to add a link to this page from table of contents.

You can preview content via OpenPublishing.Build.

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs
Comment (wtgodbe) [reply aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:18]: Example of this?

**Comment by wtgodbe** [comment]
What index/table of contents do I need to update? @Rick-Anderson ?

**Comment by wtgodbe** [comment]
Updated the sample/doc for some for some of the feedback

**Comment by Rick-Anderson** [comment]
> What index/table of contents do I need to update? @Rick-Anderson ?

#22983 I'll merge that into this branch when you're ready.

**Comment by wtgodbe** [comment]
Just merged it in, thanks!

File: aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs
Comment (Tratcher) [reply aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:20]: You've already done it with `UseHttps()`, this CertificateLoader line is now redundant and can be removed.

Comment (Tratcher) [reply aspnetcore/fundamentals/servers/kestrel/samples/6.x/Http3Sample/Program.cs:20]:
```suggestion
```

File: aspnetcore/toc.yml
Comment (Tratcher) [inline aspnetcore/toc.yml:849]: The HTTP/2 entry is just named "HTTP/2".
```suggestion
        - name: HTTP/3
```

**Review by Tratcher — APPROVED** [review]

**Comment by wtgodbe** [comment]
@Rick-Anderson any other concerns?

**Review by Rick-Anderson — APPROVED** [review]

**Comment by wtgodbe** [comment]
Thanks @Rick-Anderson! How long does it take for https://docs.microsoft.com/en-us/aspnet/core/fundamentals/?view=aspnetcore-6.0 to get updated?

**Comment by wtgodbe** [comment]
Never mind, I see it there now

**Pull request merged**
